addons/account_payment/views/account_portal_templates.xml

```diff
@@ -71,7 +71,7 @@
                     <h4 class="mb24">Pay with</h4>
                     <t t-call="payment.payment_tokens_list">
                         <t t-set="mode" t-value="'payment'"/>
-                        <t t-set="partner_id" t-value="request.env.user.partner_id.id"/>
+                        <t t-set="partner_id" t-value="invoice.partner_id.id"/>
```

Comment:
Shouldn't it be done only if the current user is the public user, like it is done on the controller?

Reply:
Maybe, but we do exactly my fix already in sale: https://github.com/odoo/odoo/blob/11.0/addons/sale_payment/views/sale_portal_templates.xml#L58